pre_processing/read_slstr_funcs.F90

```diff
@@ -326,42 +315,106 @@ subroutine slstr_resample_vis_to_tir(inarr, outarr, nx, ny, fill)
 
 end subroutine slstr_resample_vis_to_tir
 
-subroutine slstr_get_alignment(nx, ny, obnx, obny, tirlons, oblons, alignment)
+subroutine slstr_get_alignment(indir, startx, endx, sx_nad, sx_obl, &
+     ex_nad, ex_obl)
 
    use preproc_constants_m
 
    implicit none
 
-   integer,          intent(in)  :: nx
-   integer,          intent(in)  :: ny
-   integer,          intent(in)  :: obnx
-   integer,          intent(in)  :: obny
-   integer,          intent(out) :: alignment
-   real(kind=sreal), intent(in)  :: tirlons(:,:) ! (nx,ny)
-   real(kind=sreal), intent(in)  :: oblons(:,:) ! (obnx,obny)
+   character(len=*), intent(in)  :: indir
+   integer,          intent(in)  :: startx
+   integer,          intent(in)  :: endx
+   integer,          intent(out) :: sx_nad
+   integer,          intent(out) :: sx_obl
+   integer,          intent(out) :: ex_nad
+   integer,          intent(out) :: ex_obl
+
+   integer :: nx, ny, obnx, obny, alignment
+
+#ifndef CONSTANT_OBLIQUE_OFFSET
+   real(kind=sreal), allocatable :: ndlons(:,:)
+   real(kind=sreal), allocatable :: oblons(:,:)
 
    integer :: x
-   real    :: bdiff, summer
+   real    :: bdiff, total
+#endif
+
+   call get_slstr_gridsize(indir, 'in', nx, ny)
+   call get_slstr_gridsize(indir, 'io', obnx, obny)
+
+#ifdef CONSTANT_OBLIQUE_OFFSET
```

Comment:
I am very uncomfortable with this being the default option, the number of cases where `CONSTANT_OBLIQUE_OFFSET` != 548 is high enough to be troubling, which will lead to our aerosol retrieval being wrong (slightly) in those cases. Can we switch this around (or at least discuss in the next ORAC meeting) so that calculating the offset is the default option? It's not a very time-consuming piece of code anyway, so won't have a big performance impact (compared to RTTOV or the full BRDF).

Reply:
I didn't do this for efficiency reasons. From an engineering point-of-view, there *should* only be one offset and that number *should* drop out of the L0->L1 processing.

Coastlines look like shit with line-by-line code on and, to my eye, what our search does is remove the recently introduced rejigging of lines to improve collocation. I'll check the quality of the aerosol retrieval between the two modes in some validation I'm doing. Will be impressed if I get it done by Wednesday, though.

Reply:
Yes, I've not checked this with the latest baseline - so maybe it's unneeded now.
What time period of data did you use for your analysis that determined 548 was the right number? If it's not all post-Jan 2020 then any chance you coudl run something just to see how it looks now?